tools/precommit.py

```diff
@@ -31,9 +31,8 @@
 BUILDTYPES=['debug', 'release']
 NUTTXTAG = 'nuttx-7.19'
 
-# This is latest tested TizenRT commit working for IoT.js
-# Title: Merge pull request #496 from sunghan-chang/iotivity
-TIZENRT_COMMIT='0f47277170972bb33b51996a374c483e4ff2c26a'
+# TizenRT revision known to build with iotjs (Mon Sep 11 22:00:34 2017 +0900)
+TIZENRT_COMMIT='1a6d47ada5299'
```

Comment:
Have you tried to build latest TizenRT (as in: origin/master)?

It does not build, because TizenRT sources (directory layout to be precise) changed and no longer match our Make.defs.
"it is currently still under development" - for this precise reason we need to stick to revision known to work IMO. 
My take is that TizenRT should be treated same os other dependencies (tuv, jerry, http-parser) - we tag them with exact revision; they do not advance on their own.

I agree with the last sentence -- it is what we do for nuttx.

Reply:
TizenRT is updated daily and not a stable version, unlike other submodules. If you specify a commit id, we need to make the PR to update its version every day. We must handle the changing TizenRT. `precommit.py` is not for the developer environment settings. The most important purpose of this file is to test build about new PR. If build break happens, we need to know if it is a TizenRT problem, an IoT.js problem or a new PR problem of a developer. Currently, If build break occurs because of us, I think we have to make new PR to resolve it.

Reply:
I think that precommit tests should pass for master always. If we depend on external component state this would not hold. That's why we latch revisions of our dependencies. TizenRT is one of them, ergo we should control its revision as well.

Please take into consideration that *right now* we are stuck with commit ID 0f47277170972b, which points to a commit dated 'Fri Sep 1 19:07:30 2017 +0900'. In other words, IoT.js DOES NOT follow changing TizenRT.